src/IngredientList.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import './IngredientList.scss';
+
+function IngredientList({ recipe }) {
+
+    const ingredients = [];
+
+    const ingredientKeys = Object.keys(recipe).filter(k => { 
+        return k.includes('strIngredient') && recipe[k] !== ""
+    });
+
+    const numOfIngredients = ingredientKeys.length;
+
+    for (let i = 0; i < numOfIngredients; i++) {
+        ingredients.push(`${recipe[`strIngredient${i + 1}`]} - ${recipe[`strMeasure${i + 1}`]}`);
```

Comment:
Does strIngredient[n] map to strMeasure[n]?

If so, what a confusing datastructure!

I would recommend taking the recipe object, and running a do while or while loop.

inside the Loop you should push objects to array. Start at i = 1. if strRecipe1 exists, push this object to the array:
```
{ ingredient: strIngredient1, measure: strMeasure1 }
```
the loop should run until you dont find a match.

The array should be easier to work with later I think.